web/src/hooks/useTodoAPI.tsx

```diff
@@ -10,5 +12,56 @@ export function useTodoAPI() {
     setTodoAPI(new TodoAPI(token))
   }, [token])
```

Comment:
Saw this during the meeting: I don't think the `todoAPI` variable should be a `useState` with a `useEffect` to update it, it will be better with `useMemo` I think:


```typescript
const todoAPI = useMemo(() => new TodoAPI(token), [token])
```

Fewer hooks, and more importantly: No `useEffect`!

Reply:
@sebastianvitterso Im glad you pointed that out! 🙌 
Totally agree with you, but I haven't touched these lines in the refactor, and as I see it's a little out of scope? I noticed several other things that potentially could be solved better too; to write a generic fetch function, for instance. (I guess that's what @ingeridhellen mentioned below.) If I'm not wrong, the instantiation of the API lived in the components even before it was moved here (which is violation of **D** - Dependency Inversion Principle btw 👀), so I guess we're moving in the right direction! I wanted to get this refactor out rather quickly, after I had to spread fetch-logic out to the components to be able to fix #193, trying not to worsen things. I'll work on the rest as separate topics, but feel free to join in! 😸